internal/workflow_testsuite_test.go

```diff
@@ -273,14 +273,17 @@ func TestWorkflowIDUpdateWorkflowByID(t *testing.T) {
 	var suite WorkflowTestSuite
 	// Test UpdateWorkflowByID works with custom ID
 	env := suite.NewTestWorkflowEnvironment()
+	updateID := "id"
 	env.RegisterDelayedCallback(func() {
-		err := env.UpdateWorkflowByID("my-workflow-id", "update", "id", &updateCallback{
-			reject: func(err error) {
+		err := env.UpdateWorkflowByID("my-workflow-id", "update", updateID, newUpdateCallback(
+			env,
+			updateID,
```

Comment:
@Quinn-With-Two-Ns how do you feel about solution? It doesn't feel ideal that we have to pass `updateID` in 2 places, but I can't think of any other way to handle async updates. `updateCallback` needs to know what the updateID to cache its result to `env` is, and this seems like the only way to accomplish that.